apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -68,18 +78,22 @@ mod prepare_block {
         /// events
         fn build_vote_extensions_txs(
             &mut self,
-            local_last_commit: Option<ExtendedCommitInfo>,
+            local_last_commit: ExtendedCommitInfo,
         ) -> Vec<TxRecord> {
+            if is_sole_validators(&local_last_commit.votes) {
+                tracing::info!(
+                    "Sole validator and block proposer, no vote extensions to \
+                     include this round"
+                );
```

Comment:
same here

Reply:
We match here when we have a single empty vote extension from ourselves, similar to above we won't have either Ethereum events or validator set updates. I guess it raises the question of how we should handle a node restarting and launching into a round where where it needs to aggregate the validator set update vote extensions from the previous round.